src/main/java/seedu/address/model/order/Order.java

```diff
@@ -139,13 +149,28 @@ public boolean equals(Object other) {
         }
 
         Order otherOrder = (Order) other;
-        return otherOrder.getOrderId().equals(getOrderId())
+
+        boolean result = otherOrder.getOrderId().equals(getOrderId())
                 && otherOrder.getCustomerId().equals(getCustomerId())
                 && otherOrder.getCheeseType().equals(getCheeseType())
                 && otherOrder.getQuantity().equals(getQuantity())
                 && otherOrder.getOrderDate().equals(getOrderDate())
                 && otherOrder.getCompletedDate().equals(getCompletedDate())
-                && otherOrder.getCheeses().equals(getCheeses());
```

Comment:
Hmm, curious, does `equals` on the hashset not work?

Reply:
Hmmm based on the doc , it should work but when i was debugging that was giving me some problems. Will look into it again.

Reply:
I think I found out why (my part also affected by this), have to make sure that all the ids have the same hash. Will be updating in my branch